pkg/workspace/data_values_pre_processing.go

```diff
@@ -37,46 +38,89 @@ func (o DataValuesPreProcessing) Apply() (*DataValues, []*DataValues, error) {
 }
 
 func (o DataValuesPreProcessing) apply(files []*FileInLibrary) (*DataValues, []*DataValues, error) {
-	values := o.loader.schema.AsDataValue()
-	var libraryValues []*DataValues
-	for _, fileInLib := range files {
-		valuesDocs, err := o.templateFile(fileInLib)
-		if err != nil {
-			return nil, nil, fmt.Errorf("Templating file '%s': %s", fileInLib.File.RelativePath(), err)
-		}
+	_, checkSchema := o.loader.schema.(*schema.DocumentSchema)
 
-		for _, valuesDoc := range valuesDocs {
-			dv, err := NewDataValues(valuesDoc)
+	allDvs, err := o.collectDataValuesDocs(files)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// merge all Data Values YAML documents into one
+	var dvsForOtherLibraries []*DataValues
+	var dataValuesDoc *yamlmeta.Document
+	for _, dv := range allDvs {
+		switch {
+		case dv.HasLibRef():
```

Comment:
this switch statement could be made more readable as two if statements. note that type checking below switch statement would happen even if dvs are for a library ref with current impl.

```
if lib dvs {
   ...
   continue
}

if nil {
  ...
} else {
  ...
}
```

the fact that there is 2 if statements leads the reader to see that it's two unrelated pieces of functionality.